wgsl/index.bs

```diff
@@ -689,84 +682,240 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
+    /* offset(8)   align(1)  size(8)          -- implicit member alignment padding -- */
+    /* offset(16)  align(16) size(12)  */     b : vec3<f32>;
+    /* offset(28)  align(4)  size(4)   */     c : f32;
+    /* offset(32)  align(4)  size(4)   */     d : f32;
+    /* offset(36)  align(1)  size(12)         -- implicit member alignment padding -- */
+    /* offset(40)  align(8)  size(24)  */     e : A;
+    /* offset(64)  align(16) size(12)  */     f : vec3<f32>;
+    /* offset(76)  align(1)  size(4)          -- implicit member alignment padding -- */
+    /* offset(80)  align(8)  size(72)  */     g : [[stride(24)]] array<A, 3>;
+    /* offset(152) align(4)  size(4)   */     h : 32;
+    /* offset(156) align(1)  size(4)          -- implicit struct size padding -- */
+                                          };
+
+    [[group(0), binding(0)]]
+    var<storage> storage_buffer : B;
+  </xmp>
+</div>
+
+<div class='example wgsl' heading='Layout of structures with explicit member strides and alignments'>
+  <xmp highlight='rust'>
+    /*             align(8)  size(32)  */ [[block]] struct A {
+    /* offset(0)   align(4)  size(4)   */     u : f32;
+    /* offset(4)   align(4)  size(4)   */     v : f32;
+    /* offset(8)   align(8)  size(8)   */     w : vec2<f32>;
+    /* offset(16)  align(4)  size(16)  */    [[size(16)]] x: f32;
+                                          };
+
+    /*             align(16) size(208) */ [[block]] struct B {
+    /* offset(0)   align(8)  size(8)   */     a : vec2<f32>;
+    /* offset(8)   align(1)  size(8)          -- implicit field alignment padding -- */
+    /* offset(16)  align(16) size(12)  */     b : vec3<f32>;
+    /* offset(28)  align(4)  size(4)   */     c : f32;
+    /* offset(32)  align(4)  size(4)   */     d : f32;
+    /* offset(36)  align(1)  size(12)         -- implicit field alignment padding -- */
+    /* offset(48)  align(16) size(32)  */     [[align(16)]] e : A;
+    /* offset(80)  align(16) size(12)  */     f : vec3<f32>;
+    /* offset(92)  align(1)  size(4)          -- implicit field alignment padding -- */
+    /* offset(96)  align(8)  size(96)  */     g : [[stride(32)]] array<A, 3>;
+    /* offset(192) align(4)  size(4)   */     h : 32;
+    /* offset(196) align(1)  size(12)         -- implicit struct size padding -- */
+                                          };
+
+    [[group(0), binding(0)]]
+    var<uniform> uniform_buffer : B;
+  </xmp>
+</div>
+
+#### Array Layout Rules ####  {#array-layout-rules}
+
+An array element stride is the number of bytes from the start of one array
+element to the start of the next element.
+
+The first array element always has a zero byte offset from the start of the
+array.
+
+If the array type is annotated with an explicit [=stride=] decoration then this
+will be used as the array stride, otherwise the array uses an implicit stride
+equal to the size of their element type, rounded up to the alignment of the
```

Comment:
`equal to the size of the arrays element type`

Reply:
Done